ple/games/tetris_engine.py

```diff
@@ -0,0 +1,518 @@
+                     '.....'],
+                    ['.OOO.',
+                     '...O.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..O..',
+                     '.OO..',
+                     '.....',
+                     '.....']]
+
+L_SHAPE_TEMPLATE = [['...O.',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..O..',
+                     '..OO.',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '.O...',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['.OO..',
+                     '..O..',
+                     '..O..',
+                     '.....',
+                     '.....']]
+
+T_SHAPE_TEMPLATE = [['..O..',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..OO.',
+                     '..O..',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '..O..',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '.OO..',
+                     '..O..',
+                     '.....',
+                     '.....']]
+
+PIECES = {'S': S_SHAPE_TEMPLATE,
+          'Z': Z_SHAPE_TEMPLATE,
+          'J': J_SHAPE_TEMPLATE,
+          'L': L_SHAPE_TEMPLATE,
+          'I': I_SHAPE_TEMPLATE,
+          'O': O_SHAPE_TEMPLATE,
+          'T': T_SHAPE_TEMPLATE}
+
+
+class GameState:
+    def __init__(self):
+        global FPSCLOCK, DISPLAYSURF, BASICFONT, BIGFONT
+        pygame.init()
+        FPSCLOCK = pygame.time.Clock()
+        DISPLAYSURF = pygame.display.set_mode((WINDOWWIDTH, WINDOWHEIGHT))
+        BASICFONT = pygame.font.Font('freesansbold.ttf', 18)
+        BIGFONT = pygame.font.Font('freesansbold.ttf', 100)
+        pygame.display.set_caption('Tetromino')
+
+        # DEBUG
+        self.total_lines = 0
+
+        # setup variables for the start of the game
+        self.board = self.getBlankBoard()
+        self.lastMoveDownTime = time.time()
+        self.lastMoveSidewaysTime = time.time()
+        self.lastFallTime = time.time()
+        self.movingDown = False # note: there is no movingUp variable
+        self.movingLeft = False
+        self.movingRight = False
+        self.score = 0
+        self.lines = 0
+        self.height = 0
+        self.level, self.fallFreq = self.calculateLevelAndFallFreq()
+
+        self.fallingPiece = self.getNewPiece()
+        self.nextPiece = self.getNewPiece()
+
+        self.frame_step([1,0,0,0,0,0])
+
+        pygame.display.update()
```

Comment:
Why is `pygame.display.update()` being called here? PLE handles the drawing and setup of the display